pkg/eks/fargate.go

```diff
@@ -63,7 +64,26 @@ func (t *fargateProfilesTask) Do(errCh chan error) error {
 // DoCreateFargateProfiles creates fargate profiles as specified in the config
 func DoCreateFargateProfiles(ctx context.Context, config *api.ClusterConfig, fargateClient FargateClient) error {
 	clusterName := config.Metadata.Name
+
+	// Get existed farget profiles before entering loop
+	existedProfileNames, _ := fargateClient.ListProfiles(ctx)
+
 	for _, profile := range config.FargateProfiles {
+		// Check if target Fargate Profile existence before actual creation
+		var isProfileExisted bool
+		for _, existedProfileName := range existedProfileNames {
+			if profile.Name == existedProfileName {
+				logger.Info("Fargate profile %q already existed on EKS cluster %q", profile.Name, clusterName)
+				isProfileExisted = true
+				break
+			}
+		}
+
+		// Skip if target Fargate Profile have already created
+		if isProfileExisted {
+			continue
+		}
```

Comment:
Can this be extracted into its own functions?

Reply:
Done

Reply:
Also, I've linked it to issue 5652, might need to add label `kind/...`.